scripts/functional-tests.sh

```diff
@@ -333,23 +358,54 @@ wait_for_server(){
     fi
 }
 
+
+########################################
+# This function converts a given string to uppercase.
+# Arguments:
+#  $1: The string to convert to uppercase.
+# Returns:
+#  The uppercase string.
+########################################
+function to_upper() {
+    local to_convert="${1}"
+    local major_version
+    major_version=$(echo "${BASH_VERSION}" | cut -d'.' -f1)
+
+    # compatibility script to use built-in operator on bash 4+
+    # and default to tr on bash 3 or older.
```

Comment:
The project only [supports](https://github.com/instructlab/instructlab?tab=readme-ov-file#bash-version-44-or-newer) Bash 4.4. So this is not needed.

Reply:
@leseb MacOS still ships with BASH version 3, so it needs to be something that we can support out-of-the-box.

Reply:
We discussed this offline, the resolution is we will not support bash 3 and this will be dropped from the PR once the rest of this file works.